### DISCUSSION TITLE
GCP PubSub credentials file

### DISCUSSION THREAD

**Discussion by erdody** [redpanda-data/connect#2902] [Q&A]
We're evaluating using the pubsub output connector. We'd like to get GCP credentials from files (mounted k8s secrets) instead of using [env. vars.](https://docs.redpanda.com/redpanda-connect/configuration/secrets/#using-environment-variables)

We tried:
1. bloblang `file` function: the [credentials_json](https://docs.redpanda.com/redpanda-connect/components/outputs/gcp_pubsub/#credentials_json) field is not interpolated
2. [cli flags](https://docs.redpanda.com/redpanda-connect/configuration/secrets/#using-cli-flags): this doesn't seem to work with the `streams` mode, which we would also like to use

Is there any way we can achieve this currently?

If not, would you be open to a PR to add a `credentials_file` field that would just set to the corresponding option (`WithCredentialsFile`) in google's pubsub client?

**Comment by mihaitodor** [comment]
Hey @erdody 👋 I think you should be able to use the `GOOGLE_APPLICATION_CREDENTIALS` environment variable and point it to the mounted JSON file.

**Reply by erdody** [reply]
Thanks @mihaitodor, that would work if we only had one gcp connector (or all of them used the same credentials). When using streams, it would be nice to be able to control credentials per stream, individually. Would you be open to a PR to add a `credentials_file` field?

**Reply by mihaitodor** [reply]
Ahhh, that makes sense. Yes, that sounds like a good addition. Thank you!
